### PR TITLE
add Redshift Serverless e2e tests

### PR DESCRIPTION
**Pull request by GavinFrazar**
This PR adds redshift serverless discovery and IAM auth tests.

stacked on the mariadb e2e tests #40066

this also pulls a lot of the common test functions out of `rds_test.go` into `databases_test.go`

### CODE REVIEW

**Review by greedy52 — APPROVED** [review]

**Comment by GavinFrazar** [comment]
@r0mant needs codeowner approval to merge, whenever you get the chance.

I fixed some flakiness and I'm going to wait until the infra gets recreated over the weekend before I merge this anyway, just to be sure it's fixed.

**Review by r0mant — APPROVED** [review]

**Pull request merged**
